tests/kani/UnsizedCoercion/basic_inner_coercion.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// kani-flags: --mir-linker --enable-unstable
+//! Check the basic coercion from using built-in references and pointers.
+//! Tests are broken down into different crates to ensure that the reachability works for each case.
+
+mod defs;
+use defs::*;
+
+#[kani::proof]
+fn check_inner_dyn_coercion() {
+    let inner_id = kani::any();
+    let outer_id = kani::any();
+    let outer: &Outer<dyn Identity> = &Outer { inner: Inner { id: inner_id }, outer_id };
```

Comment:
I'm actually mildly surprised this compiles? It seems like it takes a reference to a temporary...

Reply:
Yes, it does and that works. I can try to find a reference explaining why.  I bumped into this another day.

Reply:
https://doc.rust-lang.org/reference/destructors.html#temporary-lifetime-extension